analyses/cellranger-analysis/submit-multiple-jobs.sh

```diff
@@ -0,0 +1,61 @@
+#!/bin/bash
+
+#BSUB -P run_cellranger
+#BSUB -J submitter
+#BSUB -q standard
+#BSUB -n 1
+#BSUB -R "rusage[mem=2500]"
+#BSUB -o submitter.out
+#BSUB -e submitter.err
+#BSUB -cwd "/research/dept/dnb/core_operations/Bioinformatics/achroni/GitHub/sc-rna-seq-snap/analyses/cellranger-analysis"
+
+########################################################################
+# Set up running directory
+prefix="/research/dept/dnb/core_operations/Bioinformatics/achroni/GitHub/sc-rna-seq-snap/analyses/cellranger-analysis"
+cd "$(dirname "${BASH_SOURCE[0]}")" 
+########################################################################
+# Create directories to save output files to
+mkdir -p ${prefix}/input
+mkdir -p ${prefix}/results
+mkdir -p ${prefix}/results/01_logs
+mkdir -p ${prefix}/results/02_cellranger_count
+mkdir -p ${prefix}/results/02_cellranger_count/DefaultParameters
+
+########################################################################
+# File in which we store the output text to verify the job execution order.
+# As the jobs will run on the cluster, we pass in a path to the directory from
+# which this launch script was run so everyone writes to the same file.
+output_file="output.txt"
+
+# Prints a simple LSF job file to standard output
+function print_job {
+    outfile=${1}
+    delay_seconds=${2}
+
+    # Preamble
+    echo "#!/bin/bash"
+
+    # Variables
+    echo "time=\`date\`"
+    echo "name=\${LSB_JOBNAME}"
+
+    # Describe job when it comes online, delay, then write completion text
+    echo "echo \"Online: name=\${name} time=\${time}\" >> ${outfile}"
+    echo "sleep ${delay_seconds}"
+    echo "echo \"Done: name=\${name}\" >> ${outfile}"
+}
+
+# Generate the LSF job files
+for ((i=1; i<=2; i++)); do
+    jobname="j${i}"
+    print_job "${output_file}" 10 > "${prefix}/${jobname}.bsub"
+    echo "bash ${prefix}/${jobname}.sh" >> "${prefix}/${jobname}.bsub"
+done
+
+# Submit job 1 - note no dependencies!
+bsub -P run_CellRanger -q standard -n 1 -R "rusage[mem=2GB]" -R "span[hosts=1]" -J j1 -o j1.out -e j1.err "bash ${prefix}/j1.bsub"
```

Comment:
When this job was submitted, the j1.out and j1.err files were not created, so instead I received an emailed job summary. Likewise the output.txt file generated in the j1.bsub script was not created. According to the emailed submission summary, the working directory for this job was /lsf_jobspool, although the correct path to my test repo was specified by the cwd and prefix paths (and all other jobs seem to have run with the correct working directory). This (and all other steps) still appear to have been run successfully though.

Reply:
@sharonfreshour That is interesting. Did the `j2.sh` job run properly once the alignments were done? Did you open the file to check if the data were inserted in there?

Reply:
@AntoniaChroni Yeah, I checked once it finished running and the j2.out and j2.err files were created for the j2 job submission (there is a typo that names the j2.out file as j1.out, but otherwise, it worked totally fine) and the j2.bsub script created/wrote to the output.txt file correctly

Reply:
Did you have `submitter.out` and `submitter.err` files instead?

Reply:
The run did create the submitter.err and submitter.out files. The j1 job tried to create the j1.out and j1.err files, but the job summary says permission was denied to create them (and to create the output.txt file for j1.bsub). I believe because that job somehow ended up with the lsf_jobspool as the working directory

Reply:
Specifying the `root_dir` as `${prefix}/j1.out` here fix the issue for properly saving the error and output files for @sharonfreshour.